utils/zerovec/src/ule/tuplevar.rs

```diff
@@ -267,22 +271,29 @@ mod tests {
                 (&vec!["dolor", "sit", "amet"]).into(),
             ),
         ];
-        let zerovec: VarZeroVec<Tuple3VarULE<str, [u8], VarZeroSlice<str>>> = (&vec).into();
+        let zerovec: VarZeroVec<Tuple3VarULE<str, [u8], VarZeroSlice<str>, Format>> = (&vec).into();
         let bytes = zerovec.as_bytes();
         let zerovec2 = VarZeroVec::parse_byte_slice(bytes).unwrap();
         assert_eq!(zerovec, zerovec2);
 
         // Test failed validation with a correctly sized but differently constrained tuple
         // Note: the str is unlikely to be a valid varzerovec
-        let zerovec3 = VarZeroVec::<Tuple3VarULE<VarZeroSlice<str>, [u8], VarZeroSlice<str>>>::parse_byte_slice(bytes);
+        let zerovec3 = VarZeroVec::<Tuple3VarULE<VarZeroSlice<str>, [u8], VarZeroSlice<str>, Format>>::parse_byte_slice(bytes);
         assert!(zerovec3.is_err());
 
         #[cfg(feature = "serde")]
         for val in zerovec.iter() {
             // Can't use inference due to https://github.com/rust-lang/rust/issues/130180
             crate::ule::test_utils::assert_serde_roundtrips::<
-                Tuple3VarULE<str, [u8], VarZeroSlice<str>>,
+                Tuple3VarULE<str, [u8], VarZeroSlice<str>, Format>,
             >(val);
         }
     }
+
+    #[test]
+    fn test_tripleule_validate() {
+        test_tripleule_validate_inner::<Index8>();
+        test_tripleule_validate_inner::<Index16>();
+        test_tripleule_validate_inner::<Index32>();
```

Comment:
Suggestion: add a test that serializing and deserializing between different formats is usually an error (unless we're unlucky and the bits are valid in both formats)

Reply:
I don't see the point of a test like that. If it succeeds, it's a valid representation. We already have scattered tests that ensure that invalid types don't parse with carefully chosen non-parsing data, I don't see any benefit to writing one around swapping formats specifically.